src/request/request.rs

```diff
@@ -1,12 +1,30 @@
-use cookie::{Cookie, CookieJar};
-use http::header::AsHeaderName;
+use cookie::CookieJar;
+use http::header::{AsHeaderName, CONTENT_LENGTH, TRANSFER_ENCODING};
 use http::request::Parts;
 use http::{HeaderMap, HeaderValue, Method, Uri, Version};
 use std::fmt::{self, Debug, Formatter};
 use std::sync::Arc;
 
 use super::param::{PathParam, PathParams, QueryParam};
-use crate::body::{HttpBody, RequestBody};
+use crate::body::{BodyReader, BodyStream, BoxBody};
+use crate::response::ResponseBuilder;
+use crate::{Error, Pipe, Response};
+
+/// The component parts of a HTTP request.
+///
+pub struct Head {
+    pub parts: Parts,
+
+    /// The cookies associated with the request. If there is not a
+    /// [CookieParser](crate::middleware::CookieParser)
+    /// middleware in the middleware stack for the request, this will be empty.
+    ///
+    cookies: CookieJar,
+
+    /// The request's path and query parameters.
+    ///
+    params: PathParams,
+}
 
 pub struct Request<T = ()> {
```

Comment:
I really like the idea of request and next being able to fit in a single cache line. The allocations required for request are likely hoisted to the top of AppService::call along with the necessary allocation for path params.